--- pf-5.5.0/src/org/pf/settings/impl/GenericNamedObject.java	2003-07-04 16:13:36.000000000 -0400
+++ pf-5.5.0~/src/org/pf/settings/impl/GenericNamedObject.java	2015-06-22 10:15:29.054081878 -0430
@@ -14,7 +14,7 @@
 // ===========================================================================
 
 /**
- *´Generic implementation for named objects
+ * Generic implementation for named objects
  *
  * @author Manfred Duchrow
  * @version 1.0
