--- pf-5.5.0/src/org/pf/reflect/ReflectUtil.java	2012-08-12 09:00:16.000000000 -0430
+++ pf-5.5.0~/src/org/pf/reflect/ReflectUtil.java	2015-06-24 21:11:36.648613083 -0430
@@ -1194,7 +1194,7 @@
 			paramTypes = new Class[0];
 		}
 
-		constructors = aClass.getDeclaredConstructors();
+		constructors = ( Constructor<T>[] ) aClass.getDeclaredConstructors();
 		for (int i = 0; i < constructors.length; i++ )
 		{
 			types = constructors[i].getParameterTypes();
